File: Best Time to Buy and Sell Stock-I.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        lowSoFar = 1000000
        for i in range(len(prices)):
            
            lowSoFar = min(lowSoFar, prices[i])
            profit = max(profit, prices[i]-lowSoFar)
            
        return profit
            
        
Time: O(N)
Space: O(1)

File: BestTimeToBuyAndSellStock-I.java
//time o(n)
//space o(1)
class Solution {
    int min = Integer.MAX_VALUE;
    public int maxProfit(int[] prices) {
        if(prices == null || prices.length == 0)
            return 0;
        int profit =0;
        for(int price : prices) {
            min = Math.min(price, min);
            profit = Math.max(profit, price-min);
        }
        return profit;
    }
}